packages/component-base/src/focus-trap-controller.js

```diff
@@ -72,6 +76,8 @@ export class FocusTrapController {
    */
   releaseFocus() {
     this.__trapNode = null;
+
+    instances.splice(instances.indexOf(this), 1);
```

Comment:
nit: I guess we could assume `this` is always the last item in the `instances` array? In that case `instances.pop()` would be enough here.
nit: untested line